Week1 [step2] iilim

iilim opened the pull request:
@Kioding
week1 step2 입니다.
스틱부분을 화면과 똑같이 하느라 공백4칸을 먼저 입력하고 난 뒤에 스틱부분을 넣었는데
이렇게 말고 다른 방법이 있는지 궁금합니다. \t 같은 경우는 스틱이 너무 오른쪽으로 치우치는 관계로 쓰지 않았습니다.
이전에 해당 폴더에 넣고 작업하라고 하셔서 이 폴더에 main.swift와 다이어그램 파일 같이 첨부하여 pr했습니다.

*문제 해결과정
요구사항대로 함수는 총 2개 이며, makeIceCream과 makeStick 이 있습니다.
함수의 이름은 동사+명사로 작명하였고, 첫번째 함수는 아이스크림 부분을 출력합니다.
예시에 나온대로 아이스크림부분은 *문자가 11개가 연속으로 이루어진 문자열이 총 8줄이 있기 때문에
print("***********)를 8번 반복출력하는 함수를 작성하였습니다.

makeStick 같은 경우는 약간의 고민이 필요했는데
가운데에 오게끔 출력하는것이 가장 중요해서
처음에는 \t을 이용했는데 너무 오른쪽으로 치우치는 바람에
이것저것 시도해본 결과 공백4칸을 먼저 입력하고 나서 | | 을 입력하면 정확하게 중앙에 오는것이 확인이 되었습니다.
스틱부분은 총 4줄이 필요하기 때문에 4번 반복하는 반복문을 작성했습니다.

순서도같은 경우는 좀 의문점이 있는게
print 함수 같은 것은 출력을 담당하는 순서도 양식을 따라야 하는건지
아니면 코드가 아닌 문장으로 작성하라고 가이드에 나와있으니 출력을 담당하는 순서도 기호가 아닌 평문 순서도 기호를 이용해야 하는지 좀 헷갈립니다. 우선 출력을 담당하는 순서도 기호가 아닌 평문 순서도 기호를 이용하여 print()부분을 작성했는데 이게 맞는지 궁금합니다.

xbskeoe293 committed:
week1 step2 1st trial

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -1,12 +1,14 @@
-//  Copyright © yagom academy. All rights reserved.
-//
+func makeIceCream() {
+    for _ in 1...8 {
```
쓰지 않는 변수는 와일드카드 식별자를 사용한 점 좋네요🥳

Kioding replied:
`1...8` 는 무슨 뜻일까요? 처음 보는 입장이라면 숫자값이 뜻하는 의미가 무엇일지 알기 어려울 것 같은데
가독성이 있는 코드를 만들려면 숫자보다는 변수를 만들어 넣어보는 건 어떨까요?😀

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -1,12 +1,14 @@
-
-print("Hello, World!")
+func makeStick() {
+    for _ in 1...4 {
```
여기도 직접 숫자값을 넣는 게 아닌 변수로 만들어볼까요?

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -1,12 +1,14 @@
-//
-//  Created by yagom.
-//  Copyright © yagom academy. All rights reserved.
-//
```
이 부분이 프로젝트 저작권 관련 부분이라면 그대로 두는 게 좋아보이네요 👀

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -1,12 +1,14 @@
 
-import Foundation
-
-print("Hello, World!")
```
필요없는 코드의 삭제 좋습니다 👍

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -1,12 +1,14 @@
+    }
+}
 
-import Foundation
```
import Foundation은 어떤 역할일까요? 지워도 괜찮을까요?

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -1,12 +1,14 @@
-//
+func makeIceCream() {
+    for _ in 1...8 {
+        print("***********")
```
직접 값을 print 하고 있는데, 이것만 보고는 무엇을 print 하고 있는지 알기 어려운데 변수로 넣어보는 건 어떨까요?

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -1,12 +1,14 @@
-print("Hello, World!")
+func makeStick() {
+    for _ in 1...4 {
+        print("    | |")
```
위 print 부분과 같은 코멘트입니다 :)

xbskeoe293 committed:
docs: week1 step2 피드백 반영
피드백 주신 사항 모두 반영하였습니다

1.우선 순서도 파일 그림파일로 변환 후에 README.md 파일을 새로 만들어서 넣었습니다
2.반복문에서 반복횟수를 담당하는 정수 부분 모두 변수처리했습니다
3.저작권 관련 부분 주석 복구하였습니다
4.import Foudation 같은 경우, c언어에서 include <*.h> 같은 역할을 하는 명령어로 판단됩니다
현재 과제 수준에서는 필요없는 명령어라 판단해서 넣지 않았고 실제로 없어도 프로그램은 정상실행됩니다
5.값을 직접 출력 하는 것이 아닌 string 변수로 선언하여 변수를 출력하게 했습니다
6.보시는 바와 같이 commit template 적용하였습니다

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -5,8 +5,24 @@
+var beforeStick: String = "    "
 
-import Foundation
+func makeIceCream() {
+    for _ in initalVal...iceCreamLength {
+        print(iceCreamRow)
+    }
+}
```
변수를 넣어 어떤 걸 위한 for문인지 알 수 있게 하셨네요!
`initalVal` 가 var로 선언되어 바뀔수도 있을까 생각되네요. 
여기서는 첫 시작은 1로 시작해도 괜찮을 것 같네요 😆

Kioding replied:
처음 보는 입장에서 `iceCreamRow` 을 보면 어떤 뜻일지 알기 어려울 것 같아요.
변수명을 어떻게 바꿔보면 좋을까요?

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -5,8 +5,24 @@
+}
 
-print("Hello, World!")
+func makeStick() {
+    for _ in initalVal...stickLength {
+        print(beforeStick+stickRow)
+    }
+}
```
`beforeStick+stickRow` 도 어떤 의미일지 알기 어려울 것 같네요.
아이스크림 막대 부분을 나타낸다는 변수명이라면 좋을 것 같습니다 👀
